pkg/envtest/helper.go

```diff
@@ -57,21 +68,20 @@ func existsUnstructured(s1, s2 []*unstructured.Unstructured) bool {
 	return false
 }
 
-func runtimeListToUnstructured(l []runtime.Object) []*unstructured.Unstructured {
+func runtimeCRDListToUnstructured(l []runtime.Object) []*unstructured.Unstructured {
 	res := []*unstructured.Unstructured{}
 	for _, obj := range l {
-		m, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj.DeepCopyObject())
-		if err != nil {
+		u := &unstructured.Unstructured{}
+		if err := crdScheme.Convert(obj, u, nil); err != nil {
```

Comment:
can avoid scheme setup entirely if we use `k8s.io/apimachinery/pkg/runtime.DefaultUnstructuredConverter`, which does the equivalent of the JSON encoder instead.

Reply:
Yeah that's what it was there before, although `DefaultUnstructuredConverter` loses TypeMeta information when converting from struct -> unstructured

Reply:
Quick test with a Cluster API object:

```
        spew.Dump(runtime.DefaultUnstructuredConverter.ToUnstructured(&v1alpha3.Cluster{}))
	out := &unstructured.Unstructured{}
	scheme.Convert(&v1alpha3.Cluster{}, out, nil)
	spew.Dump(out)
```

prints:

```
(map[string]interface {}) (len=3) {
 (string) (len=8) "metadata": (map[string]interface {}) (len=1) {
  (string) (len=17) "creationTimestamp": (interface {}) <nil>
 },
 (string) (len=4) "spec": (map[string]interface {}) (len=1) {
  (string) (len=20) "controlPlaneEndpoint": (map[string]interface {}) (len=2) {
   (string) (len=4) "host": (string) "",
   (string) (len=4) "port": (int64) 0
  }
 },
 (string) (len=6) "status": (map[string]interface {}) (len=2) {
  (string) (len=19) "infrastructureReady": (bool) false,
  (string) (len=23) "controlPlaneInitialized": (bool) false
 }
}
```

```
(*unstructured.Unstructured)(0xc000104028)({
 Object: (map[string]interface {}) (len=5) {
  (string) (len=8) "metadata": (map[string]interface {}) (len=1) {
   (string) (len=17) "creationTimestamp": (interface {}) <nil>
  },
  (string) (len=4) "spec": (map[string]interface {}) (len=1) {
   (string) (len=20) "controlPlaneEndpoint": (map[string]interface {}) (len=2) {
    (string) (len=4) "host": (string) "",
    (string) (len=4) "port": (int64) 0
   }
  },
  (string) (len=6) "status": (map[string]interface {}) (len=2) {
   (string) (len=19) "infrastructureReady": (bool) false,
   (string) (len=23) "controlPlaneInitialized": (bool) false
  },
  (string) (len=10) "apiVersion": (string) (len=25) "cluster.x-k8s.io/v1alpha3",
  (string) (len=4) "kind": (string) (len=7) "Cluster"
 }
})
```

Reply:
i've seen this too. is there a relevant apimachinery issue (or do we know if this is intended behavior?)

Reply:
https://github.com/kubernetes/kubernetes/issues/80609

Reply:
> Yeah that's what it was there before,

Lol, I can definitely read diffs :-P

I think at least in your example, it's not actually clearing it, it's just not set and it doesn't know how to populate it (since Go struct <-> gvk isn't technically 1-to-1, for some reason that I'm never managed to extract from the folks who might know).  This makes more sense then.

Reply:
Can you add these details to the PR description?

Reply:
Sure thing

Reply:
Done!